application/org.openjdk.jmc.console.agent/src/main/java/org/openjdk/jmc/console/agent/actions/AgentEditorOpener.java

```diff
@@ -0,0 +1,172 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.openjdk.jmc.console.agent.actions;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.openjdk.jmc.common.io.IOToolkit;
+import org.openjdk.jmc.console.agent.AgentJmxHelper;
+import org.openjdk.jmc.console.agent.AgentPlugin;
+import org.openjdk.jmc.console.agent.editor.AgentEditor;
+import org.openjdk.jmc.console.agent.editor.AgentEditorInput;
+import org.openjdk.jmc.console.agent.messages.internal.Messages;
+import org.openjdk.jmc.console.agent.wizards.StartAgentWizard;
+import org.openjdk.jmc.rjmx.ConnectionException;
+import org.openjdk.jmc.rjmx.IConnectionHandle;
+import org.openjdk.jmc.rjmx.IConnectionListener;
+import org.openjdk.jmc.rjmx.IServerHandle;
+import org.openjdk.jmc.rjmx.JVMSupportToolkit;
+import org.openjdk.jmc.rjmx.actionprovider.IActionFactory;
+import org.openjdk.jmc.ui.common.action.Executable;
+import org.openjdk.jmc.ui.misc.DialogToolkit;
+import org.openjdk.jmc.ui.misc.DisplayToolkit;
+
+import java.util.Objects;
+
+public class AgentEditorOpener implements IActionFactory {
+
+	@Override
+	public Executable createAction(IServerHandle serverHandle) {
+		return () -> new ConnectJob(serverHandle).schedule();
+	}
+
+	private final static class ConnectJob extends Job implements IConnectionListener {
+
+		private final IServerHandle serverHandle;
+		private IConnectionHandle connectionHandle;
+		private AgentJmxHelper helper;
+
+		private ConnectJob(IServerHandle serverHandle) {
+			super(Messages.AgentEditorOpener_JOB_NAME);
+
+			this.serverHandle = Objects.requireNonNull(serverHandle);
+		}
+
+		@Override
+		protected IStatus run(IProgressMonitor monitor) {
+			try {
+				helper = new AgentJmxHelper(serverHandle);
+				helper.addConnectionChangedListener(this);
+				connectionHandle = helper.getConnectionHandle();
+
+				IStatus ret = doRun(monitor);
+
+				helper.removeConnectionChangedListener(this);
+				return ret;
+			} catch (ConnectionException e) {
+				// FIXME: Show stacktrace? (Need to show our own ExceptionDialog in that case, or maybe create our own DetailsAreaProvider, see WorkbenchStatusDialogManager.setDetailsAreaProvider)
```

Comment:
Personally not a fan of PRs that add FIXMEs; can you provide some justification for having this? What's preventing this being done sooner rather than later?

Reply:
Fixed this to show an exception dialog, thanks